doc/decisions/operation_sequences.md

```diff
@@ -0,0 +1,253 @@
+# Operation sequences
+
+## Problem
+
+The current public API allows various kind of sequences of `kdbGet` and `kdbSet` calls.
+There is no real guideline of allowed and/or prohibited sequences, other than `kdbGet` for a certain parent key must be called before `kdbSet`:
+
+- initially after `kdbOpen`
+- after conflict errors in `kdbSet`
+
+We discovered that some sequences are currently legal but problematic for change tracking and maybe also other use cases.
+
+Currently, notification plugins assume that a `kdbSet` operation for a certain parent key always directly follows the `kdbGet` operation for that same key.
+However, this is not enforced in any way within Elektra, nor is it documented that it should.
+
+In applications, `kdbGet` and `kdbSet` of different parent keys might be interwoven in any way.
+This is problematic, as plugins may store the result of `kdbGet` and use the stored data in the following `kdbSet` to calculate a changeset.
+Plugins have to do this, as there isn't any other Elektra-provided mechanism currently to do change tracking for them.
+
+Note, that erroneous behaviour only occurs if plugins are used globally, i.e. as `notification-send` hook plugins.
+If they are mounted for specific backends (via `kdb mount`), this behaviour will not occur as plugins from different backends are isolated.
```

Comment:
```suggestion
If they are mounted for specific backends (via `kdb mount`), this behaviour will not occur as plugins from different backends are isolated and backends are treated as an atomic unit with a single parent key.
```

Just isolating them wouldn't be enough, if two calls could still have different parent keys or do different partial reads.